cmd/generate/config/rules/github.go

```diff
@@ -24,11 +29,25 @@ func GitHubPat() *config.Rule {
 		Regex:       regexp.MustCompile(`ghp_[0-9a-zA-Z]{36}`),
 		Entropy:     3,
 		Keywords:    []string{"ghp_"},
-		Allowlists:  githubAllowlist,
+		Allowlists: []*config.Allowlist{
+			{
+				Expression: `secret.substring(34) != base62encode(crc32(secret.substring(4, 34)))`,
```

Comment:
@zricethezav @bplaxco I added a _really sloppy_ expression to validate the checksum of GitHub `ghp_` tokens.

> A 32 bit checksum in the last 6 digits of each token strikes the optimal balance between keeping the random token portion at a consistent entropy and enough confidence in the checksum. We start the implementation with a CRC32 algorithm, a standard checksum algorithm. We then encode the result with a Base62 implementation, using leading zeros for padding as needed.
> https://github.blog/engineering/platform-security/behind-githubs-new-authentication-token-formats/